ruby/sdk-phase-2.md

```diff
@@ -0,0 +1,605 @@
+* Notice lack of `sleep`, we will support Ruby `sleep` via the fiber scheduler.
+* A `cancellation` can be provided to every async thing, but by default it's the workflow-level one.
+* A workflow builds a `Workflow::Definition::Info` object which contains metadata about the structure including a set of
+  `Workflow::Definition::Signal`s, `Workflow::Definition::Query`s, and `Workflow::Definition::Update`s built from the
+  handlers (but of course handlers can be added/removed at runtime via the `_handlers` hashes).
+* The `workflow_` methods above can only be called during class definition time not workflow runtime.
+* The runtime class methods above can only be called during runtime.
+* `respond_to_missing?` + `inherited` will be leveraged such that all definition-time defined handlers will return their
+  definition if they are called as class methods. So `FooWorkflow.some_signal` will return a
+  `Workflow::Definition::Signal` if an instance method of `some_signal` exists marked as a `workflow_signal`.
+  * 💭 Why? It is helpful for callers/client-side to be able to reference something more explicit than a symbol, and it
+    also helps with making sure if the name is changed we use that name instead of the method name. Symbols can always
+    be used though.
+  * ❓ Is this acceptable or too much magic?
+  * ❓ Is there a better way to reference an interaction outside of the instance of the class that works well with Ruby
+    generic typing frameworks, e.g. Sorbet?
+    * We are trying to avoid the proxy model that Java takes because we have found it is too confusing to users to hide
+      that method invocation is actually, say, activity invocation.
+
+## Dynamic Workflows, Signals, Queries, and Updates
+
+* Support dynamic pieces similar to how we have done in Python and .NET.
+* Create a new `Temporalio::Converters::RawValue` object that is a pass-through object for raw values (still subject to
+  payload codecs), and a `payload_converter` is available in workflow/activity context to convert it.
+  * Handler methods will use `RawValue` for args if told to, similar with workflow `execute`/`initialize` if
+    `workflow_raw_args` is set.
+  * `RawValue` can be returned too.
+* To mark a workflow as dynamic, `workflow_dynamic` class method is called. Only one dynamic workflow can be present for
+  a worker. The workflow can retrieve its workflow type name via `Workflow.info`
+* Dynamic signals, queries, and updates are enabled via `dynamic: true` on the calls before the methods.
+  * Dynamic handlers first argument is the string name (it's not a symbol), and all successive arguments are the actual
+    arguments.
+  * Only one dynamic handler of a type (signal, query, or update) can be on a workflow.
+
+## Workflow Asynchrony
+
+Ruby does not have a built-in task/promise/future/etc concept. However it does have fibers and we can leverage those.
```

Comment:
This looks like this is out of date and seems to contradict your `Future` stuff below?

Reply:
This is accurate. Ruby does not have a built-in concept and we can leverage fibers. We add our own future later on top of fibers of course.